jdaviz/core/tools.py

```diff
@@ -144,9 +144,31 @@ def on_update_zoom(self):
             # let's ignore and reset the tool
             return
 
-        self.viewer.state.x_min, self.viewer.state.x_max = self.interact.selected_x
+        new_interact_x = self.get_x_axis_with_aspect_ratio()
         self.viewer.state.y_min, self.viewer.state.y_max = self.interact.selected_y
 
+        # Change the x axis to make sure all y axis values from the box zoom are included
+        self.viewer.state.x_min, self.viewer.state.x_max = new_interact_x
+
+    def get_x_axis_with_aspect_ratio(self):
+        new_interact_x = self.interact.selected_x
+        # If aspect is equal, then we need to preserve that aspect ratio
+        # even if the box zoom area does not.
+        if self.viewer.state.aspect == "equal":
+            fig_axes_ratio = ((self.viewer.state.x_max - self.viewer.state.x_min) /
+                              (self.viewer.state.y_max - self.viewer.state.y_min))
+
+            # The value at index of 1 will always be the larger of the two numbers
+            x_diff = self.interact.selected_x[1] - self.interact.selected_x[0]
+            y_diff = self.interact.selected_y[1] - self.interact.selected_y[0]
+
+            if x_diff < (y_diff / (1 / fig_axes_ratio)):
+                y_with_aspect = y_diff / (1 / fig_axes_ratio)
+                new_interact_x = (self.interact.selected_x[0] - y_with_aspect / 2,
+                                  self.interact.selected_x[1] + y_with_aspect / 2)
```

Comment:
```suggestion
            if x_diff < y_diff * fig_axes_ratio:
                x_with_aspect = y_diff * fig_axes_ratio
                x_centre = (self.interact.selected_x[0] + self.interact.selected_x[1]) / 2
                new_interact_x = (x_centre - x_with_aspect / 2, x_centre + x_with_aspect / 2)
```
Squashed the multiple fractions here for clarity; but I think what you really want to resolve https://github.com/spacetelescope/jdaviz/pull/1726#issuecomment-1276552660 is to stretch the new x range symmetrically to scale with the (larger) y range.
What surprised however was that no such correction was required for `x_diff > y_diff * fig_axes_ratio` to begin with - `selected_y` seem already to be adjusted automatically. The reason behind this is apparently that
```
viewer.scales
{'x': LinearScale(allow_padding=False, max=2919.1, min=1384.2), 'y': LinearScale(max=1229.9, min=281.8)}
```
I don't know the reason for this asymmetry, but it seems you can get by without the entire post-correction above by setting instead
```python
    def _new_interact(self):
        x_scale = self.viewer.scale_x
        x_scale.allow_padding = True
        return BrushSelector(x_scale=scale_x, y_scale=self.viewer.scale_y, color=INTERACT_COLOR)
```

Reply:
Thank you for finding the cause and cleaning up the code! Yes we were wondering why the x axis seemed to work as expected but not the y. @camipacifici , I believe @dhomeier's changes fix the problem you were seeing before, at least it looks like that to me. Do you have time to check that?